examples/bpi-monolith-test-suite/bpi/components/workgroup/workgroup.ts

```diff
@@ -6,16 +6,24 @@ export class Workgroup {
      name: string;
      participants: BpiSubject[] = [];
      worksteps: Workstep[] = [];
+     workstepids: Set<string> = new Set("");
 
     constructor(name: string, id: string, owner: BpiSubject, worksteps: Workstep[],) {
-        this.name = name;
         this.id = id;
+        this.name = name;
         this.worksteps = worksteps;
         this.participants.push(owner);
     }
 
      addWorkstep(workstep: Workstep) {
-          this.worksteps.push(workstep);
+
+          //push workstep if ID is not yet in use
+          if(!(this.workstepids.has(workstep.id))) {
+               this.worksteps.push(workstep);
+          }
+
+          //add ID to running set
+          this.workstepids.add(workstep.id);
      }
 
      getWorkstepById(workstepId: string) {
```

Comment:
i know it is not related to your changes, but there can be a bug in this function if workstep with this id doesn't exist, same for  getParticipantsById, so we can quickly fix this now i think, something like this should work:
`return worksteps.length >0 && worksteps[0]`